napari/_qt/qt_main_window.py

```diff
@@ -1196,18 +1196,16 @@ def _update_theme(self, event=None):
         """Update widget color theme."""
         settings = get_settings()
         with contextlib.suppress(AttributeError, RuntimeError):
-            if event:
-                value = event.value
-                self._qt_viewer.viewer.theme = value
-                settings.appearance.theme = value
-            else:
-                value = (
-                    get_system_theme()
-                    if settings.appearance.theme == "system"
-                    else self._qt_viewer.viewer.theme
+            value = event.value if event else settings.appearance.theme
+            self._qt_viewer.viewer.theme = value
+            if value == "system":
+                # system isn't a theme, so get the name and set style sheet
+                actual_theme_name = get_system_theme()
```

Comment:
why not just 
```suggestion
                value = get_system_theme()
```

This will reduce code repetition. (but maybe having a semantic name is better).

Reply:
This was intentional, such that `viewer.theme` still reports `'system'` but the actual styling is from the actual theme.